### PR TITLE
feat: add `message` field into both `RESTPostAPIGuildForumThreads`

### PR DESCRIPTION
**Pull request by vitorlops**
**Please describe the changes this PR makes and why it should be merged:**
Note: I don't know if it's correct, if not please let me know!

**If applicable, please reference Discord API Docs PRs or commits that influenced this PR:**
- https://github.com/discord/discord-api-docs/pull/4850

### CODE REVIEW

File: deno/rest/v10/channel.ts [hunk 1]
```diff
@@ -585,7 +585,12 @@ export type RESTPostAPIChannelMessagesThreadsJSONBody = AddUndefinedToPossiblyUn
  * https://discord.com/developers/docs/resources/channel#start-thread-in-forum-channel
  */
 export type RESTPostAPIGuildForumThreadsJSONBody = RESTPostAPIChannelMessagesThreadsJSONBody &
-	RESTPostAPIChannelMessageJSONBody;
+	RESTPostAPIChannelMessageJSONBody & {
```
Comment (vladfrangu) [inline deno/rest/v10/channel.ts:588]: `RESTPostAPIChannelMessageJSONBody` is no longer needed. in the merged type. Also, for the JSON body, the type for message should be `RESTPostAPIChannelMessageJSONBody`

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: deno/rest/v10/channel.ts [hunk 2]
```diff
@@ -584,8 +584,12 @@ export type RESTPostAPIChannelMessagesThreadsJSONBody = AddUndefinedToPossiblyUn
 /**
  * https://discord.com/developers/docs/resources/channel#start-thread-in-forum-channel
  */
-export type RESTPostAPIGuildForumThreadsJSONBody = RESTPostAPIChannelMessagesThreadsJSONBody &
-	RESTPostAPIChannelMessageJSONBody;
+export type RESTPostAPIGuildForumThreadsJSONBody = RESTPostAPIChannelMessagesThreadsJSONBody & {
+	/**
+	 * First message in the forum thread
+	 */
+	message: RESTPostAPIChannelMessageFormDataBody;
```
Comment (vladfrangu) [inline deno/rest/v10/channel.ts:591]: This needs to be JSONBody not formdata body

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: deno/rest/v10/channel.ts (hunk 2 above)
Comment (vitorlops) [reply deno/rest/v10/channel.ts:591]: Oops, I misread

**Comment by vladfrangu** [comment]
Can you also update the form data so its more accurate?

File: deno/rest/v10/channel.ts [hunk 3]
```diff
@@ -594,7 +594,12 @@ export type RESTPostAPIGuildForumThreadsJSONBody = RESTPostAPIChannelMessagesThr
 /**
  * https://discord.com/developers/docs/resources/channel#start-thread-in-forum-channel
  */
-export type RESTPostAPIGuildForumThreadsFormDataBody = RESTPostAPIChannelMessageFormDataBody;
+export type RESTPostAPIGuildForumThreadsFormDataBody = {
+	/**
+	 * First message in the forum thread
+	 */
+	message: RESTPostAPIChannelMessageFormDataBody;
+} & RESTPostAPIChannelMessagesThreadsJSONBody;
```
Comment (vladfrangu) [inline deno/rest/v10/channel.ts:602]: 1. Invert the `&` (so start with the `RESTPostAPIChannelMessagesThreadsJSONBody`, and then & it with the message
2. The message should have the same type as `payload_json` from MessageFormData (ala string)

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**